packages/jaeger-ui/src/components/TracePage/TraceGraph/TraceGraph.js

```diff
@@ -0,0 +1,328 @@
+        const cdr = new DRange(m.span.startTime, m.span.startTime + m.span.duration).intersect(
+          this.getChildOfDrange(m.span.spanID)
+        );
+        return p + cdr.length;
+      }, 0);
+      const stime = ntime - childDurationsDRange;
+      const nd = {
+        count: n.members.length,
+        errors: numErrors,
+        time: ntime,
+        percent: 100 / this.props.trace.duration * ntime,
+        selfTime: stime,
+        percentSelfTime: 100 / ntime * stime,
+      };
+      // eslint-disable-next-line no-param-reassign
+      n.data = nd;
+    });
+    return traceDag;
+  }
+
+  getChildOfDrange(parentID: string): number {
+    const childrenDrange = new DRange();
+    this.getChildOfSpans(parentID).forEach(s => {
+      // -1 otherwise it will take for each child a micro (incluse,exclusive)
+      childrenDrange.add(s.startTime, s.startTime + (s.duration <= 0 ? 0 : s.duration - 1));
+    });
+    return childrenDrange;
+  }
+
+  getChildOfSpans(parentID: string): Span[] {
+    if (!this.parentChildOfMap) {
+      this.parentChildOfMap = {};
+      this.props.trace.spans.forEach(s => {
+        if (s.references) {
+          // Filter for CHILD_OF we don't want to calculate FOLLOWS_FROM (prod-cons)
+          const parentIDs = s.references.filter(r => r.refType === 'CHILD_OF').map(r => r.spanID);
+          parentIDs.forEach((pID: string) => {
+            this.parentChildOfMap[pID] = this.parentChildOfMap[pID] || [];
+            this.parentChildOfMap[pID].push(s);
+          });
+        }
+      });
+    }
+    return this.parentChildOfMap[parentID] || [];
+  }
+
+  toggleNodeMode(newMode: string) {
+    this.setState({ mode: newMode });
+  }
+
+  showHelp = () => {
+    this.setState({ showHelp: true });
+  };
+
+  closeSidebar = () => {
+    this.setState({ showHelp: false });
+  };
+
+  render() {
+    const { headerHeight, trace } = this.props;
+    const { showHelp, mode } = this.state;
+    if (!trace) {
+      return <h1 className="u-mt-vast u-tx-muted ub-tx-center">No trace found</h1>;
+    }
+
+    // Caching edges/vertices so that DirectedGraph is not redrawn
+    let ev = this.cache;
+    if (!ev) {
+      const traceDag = this.calculateTraceDag();
+      const nodes = [...traceDag.nodesMap.values()];
+      ev = convPlexus(traceDag.nodesMap);
+      ev.edges = extendFollowsFrom(ev.edges, nodes);
+      this.cache = ev;
+    }
+
+    return (
+      <div className="TraceGraph--graphWrapper" style={{ paddingTop: headerHeight + 49 }}>
+        <DirectedGraph
+          minimap
+          zoom
+          arrowScaleDampener={0}
+          className="TraceGraph--dag"
+          minimapClassName="TraceGraph--miniMap"
+          layoutManager={this.layoutManager}
+          getNodeLabel={getNodeDrawer(mode)}
+          setOnRoot={classNameIsSmall}
+          setOnEdgePath={setOnEdgePath}
+          edges={ev.edges}
+          vertices={ev.vertices}
+        />
```

Comment:
Btw, the following function has a nice effect on the edge coloring; it reduces the opacity of the edges, when zoomed out, so they don't look heavy compared to the nodes. 

https://github.com/jaegertracing/jaeger-ui/blob/f51dfd7e1eeafc11ece483c5d74fc416dac04d53/packages/jaeger-ui/src/components/TraceDiff/TraceDiffGraph/TraceDiffGraph.js#L38-L46

https://github.com/jaegertracing/jaeger-ui/blob/f51dfd7e1eeafc11ece483c5d74fc416dac04d53/packages/jaeger-ui/src/components/TraceDiff/TraceDiffGraph/TraceDiffGraph.js#L111

With the above:
<img width="472" alt="screen shot 2018-12-17 at 3 16 59 pm" src="https://user-images.githubusercontent.com/2304337/50113045-dbaf3080-020e-11e9-8c2e-ee1a23e46eed.png">


With black edges:
<img width="388" alt="screen shot 2018-12-17 at 3 16 53 pm" src="https://user-images.githubusercontent.com/2304337/50113044-db169a00-020e-11e9-8068-8e42f328a162.png">

Reply:
LMK if you don't want to add this, I'll add it to the ticket you created for enhancements.